pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook.go

```diff
@@ -121,41 +123,64 @@ func (v *fleetResourceValidator) handleMemberCluster(req admission.Request) admi
+	if strings.HasPrefix(req.Namespace, fleetMemberNamespacePrefix) {
+		// check to see if valid users other than member agent is making the request.
+		response = validation.ValidateUserForResource(req, v.whiteListedUsers)
+		// check to see if member agent is making the request only on Update.
+		if !response.Allowed && req.Operation == admissionv1.Update {
+			mcName := parseMemberClusterNameFromNamespace(req.Namespace)
+			return validation.ValidateMCIdentity(ctx, v.client, req, mcName)
+		}
+		return response
 	}
-	return validation.ValidateUserForResource(req, v.whiteListedUsers)
+	return admission.Allowed("namespace name doesn't begin with fleet-member prefix so we allow all operations on these namespaces for the request object")
 }
 
-// handleWork allows/delete the request to modify work object after validation.
-func (v *fleetResourceValidator) handleWork(ctx context.Context, req admission.Request) admission.Response {
-	if req.Operation == admissionv1.Update && isFleetMemberNamespace(req.Namespace) && req.SubResource == "status" {
+// handleEvent allows/denies request to modify event after validation.
+func (v *fleetResourceValidator) handleEvent(ctx context.Context, req admission.Request) admission.Response {
+	// hub agent creates events for MC which is cluster scoped, only member agent creates events in fleet-member prefixed namespaces.
+	if strings.HasPrefix(req.Namespace, fleetMemberNamespacePrefix) {
 		mcName := parseMemberClusterNameFromNamespace(req.Namespace)
 		return validation.ValidateMCIdentity(ctx, v.client, req, mcName)
 	}
-	return validation.ValidateUserForResource(req, v.whiteListedUsers)
+	if strings.HasPrefix(req.Namespace, fleetNamespacePrefix) || strings.HasPrefix(req.Namespace, kubeNamespacePrefix) {
+		return validation.ValidateUserForResource(req, v.whiteListedUsers)
+	}
+
+	return admission.Allowed("namespace name for this event is not a reserved namespace so we allow all operations for events on these namespaces")
 }
 
 // handlerNamespace allows/denies request to modify namespace after validation.
 func (v *fleetResourceValidator) handleNamespace(req admission.Request) admission.Response {
-	fleetMatchResult := strings.HasPrefix(req.Name, "fleet")
-	kubeMatchResult := strings.HasPrefix(req.Name, "kube")
-	if fleetMatchResult || kubeMatchResult {
+	if strings.HasPrefix(req.Name, fleetMemberNamespacePrefix) {
+		mcName := parseMemberClusterNameFromNamespace(req.Name)
+		if mcName == "" {
+			klog.V(2).InfoS("request is trying to modify a namespace called fleet-member which is not allowed",
+				"user", req.UserInfo.Username, "groups", req.UserInfo.Groups, "operation", req.Operation, "kind", req.RequestKind.Kind, "subResource", req.SubResource, "namespacedName", types.NamespacedName{Name: req.Name, Namespace: req.Namespace})
+			return admission.Denied("request is trying to modify a namespace called fleet-member which is not allowed")
+		}
 		return validation.ValidateUserForResource(req, v.whiteListedUsers)
```

Comment:
this function name probably needs to be enhanced a bit to be more readable. I can't guess what it does from its name.